core/base_service.go

```diff
@@ -295,9 +295,24 @@ func (service *BaseService) Request(req *http.Request, result interface{}) (deta
 			// Decode step was successful. Return the decoded response object in the Result field.
 			detailedResponse.Result = result
 			return
+		} else if IsTextMimeType(contentType) {
```

Comment:
It looks like this would be a breaking change for the Watson Go SDK.  In particular, for Personality Insights, where the `profile` operation can return a `text/csv` response.  The current SDK returns its result as an `io.ReadCloser` for this case:

https://github.com/watson-developer-cloud/go-sdk/blob/master/personalityinsightsv3/personality_insights_v3.go#L246

Another case is Language Translator V3, the `GetTranslatedDocument` method:

https://github.com/watson-developer-cloud/go-sdk/blob/2f7cd68ff50aac03d3801d3aa1d10a5b6d964124/languagetranslatorv3/language_translator_v3.go#L704

which could be called with `accept="text/plain"`.